docs/performance/remote-filesystem/amazon-s3.md

```diff
@@ -0,0 +1,396 @@
+
+For detailed information on constructing these classes, and the options
+provided see the [Amazon S3 documentation][s3-docs], and specifically the
+[S3 SDK documentation][s3-sdk] and [S3 ApiGen documentation][s3-api].
+
+
+#### Amazon Credentials
+
+In this example we use `CredentialProvider::defaultProvider()` to get your
+Amazon credentials.
+
+This credential provider first checks in order:
+  * Amazon specific environment variables
+  * "default" profile in ~/.aws/credentials
+  * "profile default" profile in ~/.aws/config (which is the default profile of
+    AWS CLI)
+  * Tries to make a GET Request to fetch credentials if the Ecs environment
+    variable is presented
+  * Checks for EC2 instance profile credentials
+
+For information on the `Aws\Credentials` classes, see Amazon's ApiGen
+documentation for [AWS Credentials][s3-api-creds].
+
+#### Example
+
+This is an example of the `src/Provider/AwsServiceProvider.php` file.
+
+
+```php
+<?php
+
+namespace MySiteApp\Provider;
+
+use Aws\Credentials\CredentialProvider;
+use Aws\Credentials\Credentials;
+use Aws\Sdk;
+use Silex\Application;
+use Silex\ServiceProviderInterface;
+
+class AwsServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function register(Application $app)
+    {
+        $app['aws'] = $app->share(
+            function () {
+                $options = [
+                    // The bucket's region name, e.g. 'eu-central-1'
+                    'region' => 'my-bucket-location',
+                ];
+
+                return new Sdk($options);
+            }
+        );
+
+        $app['aws.s3'] = $app->share(function ($app) {
+            $credentials = CredentialProvider::defaultProvider();
+            $options = [
+                'version' => '2006-03-01',
+                'credentials' => $credentials,
+            ];
+
+            return $app['aws']->createS3($options);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function boot(Application $app)
+    {
+    }
+}
+```
+
+Optionally, you may with to construct your own `\Aws\Credentials\Credentials`
+object, and pass that to `CredentialProvider::fromCredentials()`, e.g.
+
+```php
+        $app['aws.s3'] = $app->share(function ($app) {
+            $accessKey = 'your access key from Amazon';
+            $secretKey = 'your secret key from Amazon';
+            $credentials = CredentialProvider::fromCredentials(new Credentials($accessKey, $secretKey));
+            $options = [
+                'version' => '2006-03-01',
+                'credentials' => $credentials,
+            ];
+
+            return $app['aws']->createS3($options);
+        });
+```
+
+<p class="note"><strong>Note:</strong> This particular exacmple should be
+considered conceptual, to simply illustrate the basic requirements.
+<br>
+Saving of thngs like secrect keys to code that gets committed to version
+control is bad security practice.
+</p>
+
+
+### Assets
```

Comment:
I would move this below the filesystem section. Filesystem is the next logical jump from AWS configuration.